.github/workflows/release.yml

```
@@ -26,6 +26,6 @@ jobs:
     - name: Build
       run: python -m build .
     - name: Publish
-      uses: pypa/gh-action-pypi-publish@master
+      uses: pypa/gh-action-pypi-publish@release/v1
       with:
         skip_existing: true
```

Also, the options are now kebab-cased:
```suggestion
        skip-existing: true
```

Though, I wouldn't advertise it, as it's a bad practice to have uncoordinated parallel publishing.

---

Since this workflow runs on pushes to `main` rather than on release, I think this is necessary, but agree that this should be reworked to publish on tag/release.

---

> I think this is necessary

It's a rather dangerous assumption. Pretty sure this can result in misleading situations when people notice something got published with no way to trace which job did this as there would be a number of new commits before people realize that one of those jobs succeeded publishing something. This is going to especially surprising when there's no automatic version generation integrated with something like setuptools-scm.

---

Hence, I'd prefer this to fail loudly, so people get emails about failing jobs and are motivated to investigate.